pynng/nng.py

```diff
@@ -908,3 +989,103 @@ def close(self):
         check_err(lib.nng_pipe_close(self.pipe))
         self._closed = True
 
+
+class Message:
+    """
+    Python interface for nng_msg.  See
+    https://nanomsg.github.io/nng/man/tip/nng_msg.5.html
+
+    Messages are immutable.
+
+    Warning:
+
+        Access to the message's underlying data buffer can be accessed with the
+        ``buffer`` attribute.  However, care must be taken not to send a message
+        while a reference to the buffer is still alive; if the buffer is used after
+        a message is sent, a segfault or data corruption will result.
+
+    """
+
+    def __init__(self, data, pipe=None):
+        self._pipe = pipe
+        # NB! There are two ways that a user can free resources that an nng_msg
+        # is using: either sending with nng_sendmsg (or the async equivalent)
+        # or with nng_msg_free.  We don't know how this msg will be used, but
+        # we need to **ensure** that we don't try to double free.  So the only
+        # way to send a message is with the send() and asend() methods on the
+        # object.
+        self._mem_freed = False
+        self._mem_freed_lock = threading.Lock()
+
+        if isinstance(data, ffi.CData) and \
+                ffi.typeof(data).cname == 'struct nng_msg *':
+            self._nng_msg = data
+        else:
+            msg_p = ffi.new('nng_msg **')
+            check_err(lib.nng_msg_alloc(msg_p, 0))
+            msg = msg_p[0]
+            check_err(lib.nng_msg_append(msg, data, len(data)))
+            self._nng_msg = msg
+
+    @property
+    def pipe(self):
+        return self._pipe
+
+    @pipe.setter
+    def pipe(self, pipe):
+        if pipe is None:
+            check_err(lib.nng_msg_set_pipe(self._nng_msg, ffi.NULL))
+            self._pipe = None
+            return
+        if not isinstance(pipe, Pipe):
+            msg = 'pipe must be type Pipe, not {}'
```

Comment:
` msg = 'pipe must be type Pipe, not {}'.format(type(pipe))`

I mean heck you could do it all in the `ValueError` arg.

Reply:
+1